rust-runtime/aws-smithy-types/src/body.rs

```diff
@@ -135,9 +140,14 @@ impl SdkBody {
         }
     }
 
-    fn poll_inner(
+    pub(crate) async fn next(&mut self) -> Option<Result<Bytes, Error>> {
+        let mut me = Pin::new(self);
+        poll_fn(|cx| me.as_mut().poll_next(cx)).await
+    }
+
+    pub(crate) fn poll_next(
```

Comment:
for `poll_next` and the `is_end_stream` methods below, I wonder if it might be better to define them in another impl block and feature gate the impl block like so:

```rust
#[cfg(any(
    feature = "feature-a",
    feature = "feature-b",
    feature = "feature-c",
    feature = "feature-d"
))]
fn my_fn() {
    todo!();
}
```

That way, we don't render it if it's unusable and we avoid having to turn of lints for specific lines.

Reply:
_(Although, I guess if we always have the `Once` variant then this suggestion doesn't work.)_

Reply:
Glad to hear you also consider having feature-gated code in a separate module (because I did during development). We will definitely have a new feature-gate when `http-body and `hyper` goes to 1.0.x, so moving `http-body-0-4-x` specific code to separate modules makes sense to me. Addressed in 70ab9d2.